synapse/rest/client/versions.py

```diff
@@ -31,6 +32,18 @@ def __init__(self, hs):
         super(VersionsRestServlet, self).__init__()
         self.config = hs.config
 
+        # Calculate these once since they shouldn't change after start-up.
+        self.e2ee_forced_public = (
+            RoomCreationPreset.PUBLIC_CHAT
+            in self.config.encryption_enabled_by_default_for_room_presets
+        )
+        self.e2ee_forced_private = (
+            RoomCreationPreset.PRIVATE_CHAT
+            in self.config.encryption_enabled_by_default_for_room_presets
+            or RoomCreationPreset.TRUSTED_PRIVATE_CHAT
```

Comment:
I wondered if this should be `and` instead of `or`...

Reply:
I think neither? https://github.com/matrix-org/synapse/blob/837293c314b47e988fe9532115476a6536cd6406/docs/sample_config.yaml#L2176-L2191 has options of `all`, `invite`, or `off` (default)

Reply:
In the case `invite`, the config ends up being `["private_chat", "trusted_private_chat"]`.

All is the two above + `"public_chat"`.

(Off it ends up being empty: `[]`.)

Reply:
exciting. Probably `and` in that case, for added safety

Reply:
Or we can add a third item as `trusted_private`?

Reply:
sure, whatever is easiest :D

Reply:
I separated it out, this should let the client tweak behavior without the server needing changes.